threedi_modelchecker/checks/cross_section_definitions.py

```diff
@@ -319,3 +319,110 @@ def get_invalid(self, session):
 
     def description(self):
         return f"{self.column_name} should be monotonically increasing for open YZ profiles. Perhaps this is actually a closed profile?"
+
+
+class CrossSectionMinimumDiameterCheck(CrossSectionBaseCheck):
+    """Check if cross section widths and heights are large enough"""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(column=models.CrossSectionDefinition.id, *args, **kwargs)
+
+    def configuration_type(
+        self, shape, first_width, last_width, first_height, last_height
+    ):
+        if (
+            (shape in [0, 2, 3, 8])
+            or (shape in [5, 6] and last_width == 0)
+            or (
+                shape == 7
+                and ((first_width, first_height) == (last_width, last_height))
+            )
+        ):
+            return "closed"
+        elif (
+            (shape == 1)
+            or (shape in [5, 6] and last_width > 0)
+            or (
+                shape == 7
+                and ((first_width, first_height) != (last_width, last_height))
+            )
+        ):
+            return "open"
+        else:
+            return "open"
+
+    def get_heights(self, shape, widths, heights):
+        if shape == 2:
+            heights = widths
+        if shape in [3, 8]:
+            heights = [1.5 * i for i in widths]
+
+        first_width = widths[0]
+        last_width = widths[-1]
+
+        if shape == 7:
+            max_width = round((max(widths) - min(widths)), 9)
+        else:
+            max_width = max(widths)
+
+        if heights:
+            first_height = heights[0]
+            last_height = heights[-1]
+            if shape == 7:
+                max_height = round((max(heights) - min(heights)), 9)
+            else:
+                max_height = max(heights)
+        else:
+            first_height = None
+            last_height = None
+            max_height = None
+
+        return first_height, last_height, max_height, first_width, last_width, max_width
+
+    def get_invalid(self, session):
+        invalids = []
+        for record in self.to_check(session).filter(
+            (models.CrossSectionDefinition.width != None)
+            & (models.CrossSectionDefinition.width != "")
+        ):
+            try:
+                widths = [float(x) for x in record.width.split(" ")]
+                heights = (
+                    [float(x) for x in record.height.split(" ")]
+                    if record.height not in [None, ""]
+                    else []
+                )
+            except ValueError:
+                continue  # other check catches this
+
+            (
```

Comment:
Suggestion here is to split along lines of the cross section shape enum:

"dispatch" pattern with function signature

```func(widths: List[float], heights: List[float]) -> float, Optional[float
```
```
if (record.shape.value) == CrossSectionShape.CIRCLE:
    max_width, max_height = self.get_max_dimensions_circle(widths, heights)
elif ...

````